File: java/src/solutions/1848_minimum_distance_to_the_target_element.java
class Solution {

  public int getMinDistance(int[] nums, int target, int start) {
    int min = Integer.MAX_VALUE;
    for (int i = 0; i < nums.length; i++) {
      if (target == nums[i]) {
        int local = Math.abs(i - start);
        if (local < min) {
          min = local;
        } else {
          break;
        }
      }
    }
    return min;
  }
}

File: rust/src/bin/1848_minimum_distance_to_the_target_element.rs
struct Solution {}
impl Solution {
    pub fn get_min_distance(nums: Vec<i32>, target: i32, start: i32) -> i32 {
        let start = start as usize;
        let mut i = 0_usize;
        loop {
            let left = start.checked_sub(i).and_then(|idx| nums.get(idx)) == Some(&target);
            let right = nums.get(start + i) == Some(&target);
            if left || right {
                return i as i32;
            }
            i += 1;
        }
    }
}

fn main() {}
